src/main/kotlin/com/sourcegraph/cody/config/migration/SettingsMigration.kt

```diff
@@ -54,16 +54,19 @@ class SettingsMigration : Activity {
       organiseChatsByAccount(project)
     }
     RunOnceUtil.runOnceForApp("CodyApplicationSettingsMigration") { migrateApplicationSettings() }
-    RunOnceUtil.runOnceForApp("ToggleCodyToolWindowAfterMigration") {
+    RunOnceUtil.runOnceForProject(project, "ToggleCodyToolWindowAfterMigration") {
       toggleCodyToolbarWindow(project)
     }
-    RunOnceUtil.runOnceForApp("CodyAccountsIdsRefresh") { refreshAccountsIds(project) }
-    RunOnceUtil.runOnceForApp("CodyAssignOrphanedChatsToActiveAccount") {
+    RunOnceUtil.runOnceForProject(project, "CodyAccountsIdsRefresh") { refreshAccountsIds(project) }
+    RunOnceUtil.runOnceForProject(project, "CodyAssignOrphanedChatsToActiveAccount") {
```

Comment:
I see the change from `runOnceForApp` to `runOnceForProject` (and that `CodyMigrateChatHistory` is for project too). 

AFAIK Cody state is persisted between projects. In particular, the logged in account remains unchanged when switching project while still using the same IDE (e.g. IntelliJ Ultimate). 

On the other hand `CodyAuthenticationManager` is a project level service. This is somewhat inconsistent. 

Anyway, I think this change is harmless. Eventually, all of these should work.

Reply:
FYI conversations are also saved per project.